src/UI/Components/Diagram/Context/ComposerEditorProvider.tsx

```diff
@@ -115,37 +120,54 @@ export const ComposerEditorProvider: React.FC<Props> = ({
       relatedInventoriesQuery.error.message,
     );
     const retry = relatedInventoriesQuery.refetch;
-    const ariaLabel = "ComposerEditor-relatedInventoriesQuery_failed";
+    const ariaLabel = "ComposerEditorProvider-RelatedInventoriesQuery_failed";
 
     return renderErrorView(message, ariaLabel, retry);
   }
 
-  if (serviceModelsQuery.isSuccess && instanceWithRelationsQuery.isSuccess) {
-    if (!mainService) {
-      const message = words("instanceComposer.noServiceModel.errorMessage")(
-        serviceName,
-      );
-      const retry = serviceModelsQuery.refetch;
-      const ariaLabel = "ComposerEditor-NoMainService_failed";
-
-      return renderErrorView(message, ariaLabel, retry);
-    }
+  if (serviceModelsQuery.isSuccess && !mainService) {
+    const message = words("instanceComposer.noServiceModel.errorMessage")(
+      serviceName,
+    );
+    const retry = serviceModelsQuery.refetch;
+    const ariaLabel = "ComposerEditorProvider-NoServiceModel_failed";
 
+    return renderErrorView(message, ariaLabel, retry);
+  }
+
+  if (serviceModelsQuery.isSuccess && instanceWithRelationsQuery.isSuccess) {
     return (
       <InstanceComposerContext.Provider
         value={{
-          mainService,
+          mainService: mainService as ServiceModel, // there is no possibility to instanceWithRelationsQuery be in success state without mainService, and there is assertion above the if services are fetch but there is no mainService to display errorView
```

Comment:
Won't adding the check for undefined in the if statement resolve the need for casting?